controllers/kuadrant_controller.go

```diff
@@ -479,3 +502,58 @@ func (r *KuadrantReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&authorinov1beta1.Authorino{}).
 		Complete(r)
 }
+
+func (r *KuadrantReconciler) isConsolePluginAvailable() bool {
```

Comment:
It is already implemented in `kuadrantgatewayapi.IsCRDInstalled`
I patched it and the outcome is:
```diff
diff --git a/controllers/kuadrant_controller.go b/controllers/kuadrant_controller.go
index b7c89dd..7d7589f 100644
--- a/controllers/kuadrant_controller.go
+++ b/controllers/kuadrant_controller.go
@@ -177,7 +177,13 @@ func (r *KuadrantReconciler) Reconcile(eventCtx context.Context, req ctrl.Reques
                return statusResult, nil
        }
 
-       if r.isConsolePluginAvailable() {
+       consolePluginGVK := consolev1.GroupVersion.WithKind("ConsolePlugin")
+       isConsolePluginAvailable, err := kuadrantgatewayapi.IsCRDInstalled(r.RestMapper, consolePluginGVK.Group, consolePluginGVK.Kind, consolePluginGVK.Version)
+       if err != nil {
+               return ctrl.Result{}, err
+       }
+
+       if isConsolePluginAvailable {
                logger.Info("ConsolePlugin API is available, proceeding to apply manifest")
                if err := r.applyManifest(ctx, "bundle/manifests/kuadrant-console-plugin.yaml", r.Client()); err != nil {
                        logger.Error(err, "failed to apply manifest")
@@ -501,17 +507,6 @@ func (r *KuadrantReconciler) SetupWithManager(mgr ctrl.Manager) error {
                Complete(r)
 }
 
-func (r *KuadrantReconciler) isConsolePluginAvailable() bool {
-       gvk := consolev1.GroupVersion.WithKind("ConsolePlugin")
-       mapping, err := r.RestMapper.RESTMapping(gvk.GroupKind(), gvk.Version)
-       if err != nil || mapping == nil {
-               log.Log.Info("ConsolePlugin API is not available")
-               return false
-       }
-       log.Log.Info("ConsolePlugin API is available")
-       return true
-}
-
 func (r *KuadrantReconciler) applyManifest(ctx context.Context, filePath string, k8sClient client.Client) error {
        logger := logr.FromContextOrDiscard(ctx)
```